src/libvfcfuncinstrument/libVFCFuncInstrument.cpp

```diff
@@ -0,0 +1,416 @@
+                                          &have_double,
+                                          M);
+
+              // If not, don't instrument the function
+              if (!have_double && !have_float)
+                continue;
+
+              // Demangle the name of F //
+              Name = demangle(Name);
+
+              // Create function ID
+              std::string FunctionName = File + "/" + Name + "_" + std::to_string(Line);
+              Value *FunctionID = Builder.CreateGlobalStringPtr(FunctionName);
+ 
+              // Constants creation 
+              Constant * isLibraryFunction = ConstantInt::get(Int8Ty, is_from_library);
+              Constant * isInstrinsicFunction = ConstantInt::get(Int8Ty, is_intrinsic);
+              Constant * haveFloat = ConstantInt::get(Int8Ty, have_float);
+              Constant * haveDouble = ConstantInt::get(Int8Ty, have_double);
+              Constant * zero_const = ConstantInt::get(Int32Ty, 0);
+
+              // Enter function arguments
+              std::vector<Value *> EnterArgs;
+              EnterArgs.push_back(FunctionID);
+              EnterArgs.push_back(isLibraryFunction);
+              EnterArgs.push_back(isInstrinsicFunction);
+              EnterArgs.push_back(haveFloat);
+              EnterArgs.push_back(haveDouble);
+              EnterArgs.push_back(zero_const);
+
+              // Exit function arguments
+              std::vector<Value *> ExitArgs;
+              ExitArgs.push_back(FunctionID);
+              ExitArgs.push_back(isLibraryFunction);
+              ExitArgs.push_back(isInstrinsicFunction);
+              ExitArgs.push_back(haveFloat);
+              ExitArgs.push_back(haveDouble);
+              ExitArgs.push_back(zero_const);
+
+              // Temporary function call to make loads and stores insertions easier
+              CallInst * enter_call = CallInst::Create(func_enter, EnterArgs);
+              enter_call->insertBefore(pi);
+
+              // The clone of the instrumented function's call will be used with hooked arguments
+              // and the original one will be replaced by the load of the hooked return value
+              Instruction * func_call = pi->clone();
+              func_call->insertBefore(pi);
+
+              // Temporary function call to make loads and stores insertions easier
+              CallInst * exit_call = CallInst::Create(func_exit, ExitArgs);
+              exit_call->insertBefore(pi);
+
+              // Instrumented function arguments
+              std::vector<Value *> FunctionArgs;
+
+              int m = 0;
+              // loop over arguments
+              for (auto it = pi->op_begin(); it < pi->op_end()-1; it++){
+                // Get Value
+                Value *v = (*it);
+
+                // Get type
+                Type* VType;
+                Value *TypeName;
+
+                if (v->getType() == FloatTy){
+                  VType = FloatTy;
```

Comment:
You can hoist out of the if-cases VType, and write just once VType = v->getType().